tracer/transport.go

```diff
@@ -145,6 +164,19 @@ func (t *httpTransport) SendServices(services map[string]Service) (*http.Respons
 	if err != nil {
 		return &http.Response{StatusCode: 0}, err
 	}
+	defer func() {
+		// The default HTTP client's Transport does not
+		// attempt to reuse HTTP/1.0 or HTTP/1.1 TCP connections
+		// ("keep-alive") unless the Body is read to completion and is
+		// closed.
+		// Buffer the response body so the caller doesn't need to worry about
+		// reading and closing the response. This isn't very expensive because
+		// the responses from the Agent are always short.
+		var buf bytes.Buffer
+		io.Copy(&buf, response.Body)
```

Comment:
@palazzem the `io.Copy` will copy all of the bytes in the response to the buffer. The `ioutil.NopCloser` merely adds an empty `Close` method to an `io.Reader`. This is necessary because the `Body` field on the `ResponseWriter` is an `io.ReadCloser`.